CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.8)
+
+set(CMAKE_CXX_STANDARD 14)
+
+project (SketchUpApiCppWrapper)
+
+# Define various paths:
+
+set(CPP_API_BASENAME "SUAPI-CppWrapper")
+set(CPP_API_INCLUDE_PATH "${PROJECT_SOURCE_DIR}/include")
+set(CPP_API_SOURCE_PATH "${PROJECT_SOURCE_DIR}/src")
+set(CPP_API_TESTS_PATH "${PROJECT_SOURCE_DIR}/tests")
+
+set(CPP_API_THIRD_PARTY_PATH "${PROJECT_SOURCE_DIR}/third-party")
+
+# TODO(thomthom): Configure for Mac:
+set(SLAPI_PATH "${CPP_API_THIRD_PARTY_PATH}/slapi")
+set(SLAPI_INCLUDE_PATH "${SLAPI_PATH}/win/headers")
+set(SLAPI_BINARIES_PATH "${SLAPI_PATH}/win/binaries")
+set(SLAPI_BINARIES_X64_PATH "${SLAPI_BINARIES_PATH}/sketchup/x64")
+# TODO(thomthom): Add library paths per platform, and try to add just "SketchUpAPI"
+#                 to `target_link_libraries`.
+set(SLAPI_LIB "${SLAPI_BINARIES_X64_PATH}/SketchUpAPI.lib")
+
+set(GOOGLETEST_PATH "${CPP_API_THIRD_PARTY_PATH}/googletest/googletest")
+set(GOOGLETEST_INCLUDE_PATH "${GOOGLETEST_PATH}/include")
+
+
+# GoogleTest 1.8 produce a warning that must be silenced. The fix is yet to be
+# pushed to the GoogleTest stable branch.
+#   warning C4996: 'std::tr1': warning STL4002: The non-Standard std::tr1
+#   namespace and TR1-only machinery are deprecated and will be REMOVED. You can
+#   define _SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING to acknowledge that you
+#   have received this warning.
+add_definitions(-D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING)
```

Comment:
Do we really want this to apply to all of the folowing targets, or just GoogleTest?  It should be appended to the GoogleTest.cmake file and made so it only acts on GoogleTest.
```cmake
...
add_library(GoogleTest STATIC third-party/googletest/googletest/src/gtest-all.cc)
target_compile_definitions(GoogleTest PRIVATE -D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING)
...
```

Reply:
It could probably be omitted for the wrapper lib, but the Tests project also needs it as including the GoogleTest headers cause warnings also there.